azure-pipelines.yml

```diff
@@ -13,6 +13,14 @@ steps:
 
 - script: |
     echo Configuring Environment
+    echo "y" | $ANDROID_HOME/tools/bin/sdkmanager --install 'system-images;android-27;google_apis;x86'
+    echo "no" | $ANDROID_HOME/tools/bin/avdmanager create avd -n xamarin_android_emulator -k 'system-images;android-27;google_apis;x86' --force
+    echo $ANDROID_HOME/emulator/emulator -list-avds
+    echo "Starting emulator"
+    nohup $ANDROID_HOME/emulator/emulator -avd xamarin_android_emulator -no-snapshot > /dev/null 2>&1 &
+    $ANDROID_HOME/platform-tools/adb wait-for-device shell 'while [[ -z $(getprop sys.boot_completed | tr -d '\r') ]]; do sleep 1; done; input keyevent 82'
```

Comment:
checking `sys.boot_completed` might sometimes be not reliable if CI is too slow, because the package manager might take more time to start. I'd rather verify if `adb shell pm list packages` returns non-zero exit code

Reply:
Yeah I accept that. I see we do that at many places but I went with azure suggestions and so far their CI is faster. Travis took more than 20mins to complete but azure agents took only 10mins. Will definitely check and improve if required.

Reply:
Travis is so slow because their VMs don't support KVM virtualization required for x86 emulator. And, of course, emulating ARM instructions set on Intel processors is slow

Reply:
True